packages/edit-site/src/components/global-styles/typography-panel.js

```diff
@@ -101,6 +102,15 @@ function useHasTextTransformControl( name, element ) {
 	return supports.includes( 'textTransform' );
 }
 
+function useHasTextDecorationControl( name, element ) {
+	// This is an exception for link elements.
+	// We shouldn't allow other blocks or elements to set textDecoration
+	// because this will be inherited by their children.
+	if ( ! name && element === 'link' ) {
+		return true;
+	}
```

Comment:
Minor, but this hook returns `undefined` when `false` would be more appropriate.

```suggestion
	return ! name && element === 'link';
```